cranelift/filetests/src/test_compile.rs

```diff
@@ -50,25 +52,36 @@ impl SubTest for TestCompile {
             comp_ctx.func.display(isa)
         );
 
-        // Verify that the returned code size matches the emitted bytes.
-        let mut sink = SizeSink { offset: 0 };
-        binemit::emit_function(
-            &comp_ctx.func,
-            |func, inst, div, sink, isa| isa.emit_inst(func, inst, div, sink),
-            &mut sink,
-            isa,
-        );
-
-        if sink.offset != total_size {
-            return Err(format!(
-                "Expected code size {}, got {}",
-                total_size, sink.offset
-            ));
+        if !isa.get_mach_backend().is_some() {
+            // Verify that the returned code size matches the emitted bytes.
+            let mut sink = SizeSink { offset: 0 };
+            binemit::emit_function(
+                &comp_ctx.func,
+                |func, inst, div, sink, isa| isa.emit_inst(func, inst, div, sink),
+                &mut sink,
+                isa,
+            );
+
+            if sink.offset != total_size {
+                return Err(format!(
+                    "Expected code size {}, got {}",
+                    total_size, sink.offset
+                ));
+            }
+
+            // Run final code through filecheck.
+            let text = comp_ctx.func.display(Some(isa)).to_string();
+            run_filecheck(&text, context)
+        } else {
+            let disasm = comp_ctx
+                .mach_compile_result
```

Comment:
I hadn't noticed this `mach_compile_result` before and this comment is irrelevant to this PR, but it does feel like there is probably a re-factoring out there somewhere where the new backend and the old co-exist a little more cleanly (e.g. have the Mach logic in `DisplayFunction` and `TargetIsa` instead of a separate top-level field or something along those lines).

Reply:
Yeah, in general I agree; it's a little tricky because the interfaces don't provide quite the same functionality, but I'll think about this a bit more to see if there isn't some way to clean it up.